probe/kubernetes/client.go

```diff
@@ -0,0 +1,91 @@
+package kubernetes
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// These constants are keys used in node metadata
+const (
+	Namespace = "kubernetes_namespace"
+)
+
+// Client keeps track of running kubernetes pods and services
+type Client interface {
+	Stop()
+	WalkPods(f func(Pod) error) error
+	WalkServices(f func(Service) error) error
+}
+
+type client struct {
+	quit             chan struct{}
+	client           cache.Getter
```

Comment:
If it fails when listing or resyncing, it will retry every resync period (default: 10s). If it fails when trying to watch, it will retry every second.